internal/provider/context_field_resource.go

```diff
@@ -0,0 +1,320 @@
+	}
+
+	if api.Stickiness != nil {
+		m.Stickiness = types.BoolValue(*api.Stickiness)
+	} else {
+		m.Stickiness = types.BoolNull()
+	}
+
+	var legalValues []attr.Value
+
+	if api.LegalValues != nil {
+		legalValueElements := make([]attr.Value, 0, len(api.LegalValues))
+		for _, legalValue := range api.LegalValues {
+			if legalValue.Description != nil {
+				legalValueElements = append(legalValueElements, types.ObjectValueMust(
+					map[string]attr.Type{
+						"value":       types.StringType,
+						"description": types.StringType,
+					},
+					map[string]attr.Value{
+						"value":       types.StringValue(legalValue.Value),
+						"description": types.StringValue(*legalValue.Description),
+					},
+				))
+			} else {
+				legalValueElements = append(legalValueElements, types.ObjectValueMust(
+					map[string]attr.Type{
+						"value":       types.StringType,
+						"description": types.StringType,
+					},
+					map[string]attr.Value{
+						"value":       types.StringValue(legalValue.Value),
+						"description": types.StringNull(),
+					},
+				))
+			}
+		}
+		legalValues = legalValueElements
+	} else {
+		legalValues = []attr.Value{}
+	}
+
+	m.LegalValues = types.ListValueMust(
+		types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				"value":       types.StringType,
+				"description": types.StringType,
+			},
+		},
+		legalValues,
+	)
+}
+
+func populateContextField(request ContextFieldSetter, plan contextFieldResourceModel) error {
+	if !plan.Description.IsNull() {
+		request.SetDescription(*plan.Description.ValueStringPointer())
+	}
+
+	if !plan.Stickiness.IsNull() {
+		request.SetStickiness(*plan.Stickiness.ValueBoolPointer())
+	} else {
+		request.SetStickiness(false)
+	}
+
+	if !plan.LegalValues.IsNull() && !plan.LegalValues.IsUnknown() {
+		legalValuesList, diags := plan.LegalValues.ToListValue(context.Background())
+		if diags.HasError() {
+			return fmt.Errorf("error extracting legal values: %s", diags)
+		}
+
+		var legalValues []unleash.LegalValueSchema
+
+		for _, legalValue := range legalValuesList.Elements() {
+			objValue, ok := legalValue.(basetypes.ObjectValue)
+			if !ok {
+				return fmt.Errorf("non-object value in legal_values: %s", diags)
+			}
+
+			valueAttr := objValue.Attributes()["value"]
+			descriptionAttr := objValue.Attributes()["description"]
+
+			legalValueSchema := unleash.LegalValueSchema{}
+
+			if stringValue, ok := valueAttr.(types.String); ok && !stringValue.IsNull() && !stringValue.IsUnknown() {
+				legalValueSchema.Value = stringValue.ValueString()
+			}
+
+			if stringValue, ok := descriptionAttr.(types.String); ok && !stringValue.IsNull() && !stringValue.IsUnknown() {
+				legalValueSchema.Description = stringValue.ValueStringPointer()
+			}
+
+			legalValues = append(legalValues, legalValueSchema)
+		}
+
+		request.SetLegalValues(legalValues)
+	} else {
+		request.SetLegalValues([]unleash.LegalValueSchema{})
+	}
+	return nil
+}
+
+type ContextFieldSetter interface {
```

Comment:
This is a sneaky little trick where all the api models are different types but they do have these methods in common. So we can let Go's structural typing to pretend they're all the same so I don't have to rewrite the handling code three times